### PR TITLE
fix: minus operator resType

### PR DESCRIPTION
**Pull request by facetosea**
# Description

Please briefly describe the code changes in this pull request.

# Checklist

Please check the items in the checklist if applicable.

- [ ] Is the user manual updated?
- [ ] Are the test cases passed and automated?
- [ ] Is there no significant decrease in test coverage?

### CODE REVIEW

File: source/libs/scalar/src/filter.c [hunk 1]
```diff
@@ -3831,13 +3849,16 @@ int32_t fltInitFromNode(SNode *tree, SFilterInfo *info, uint32_t options) {
...
-  SFltBuildGroupCtx tctx = {.info = info, .group = group};
+  SFltBuildGroupCtx tctx = {.info = info, .group = group, .ignore = false};
   nodesWalkExpr(tree, fltTreeToGroup, (void *)&tctx);
   if (TSDB_CODE_SUCCESS != tctx.code) {
     taosArrayDestroyEx(group, filterFreeGroup);
     code = tctx.code;
     goto _return;
   }
+  if (tctx.ignore) {
+    FILTER_SET_FLAG(info->status, FI_STATUS_EMPTY);
```
Comment (dapan1121) [inline source/libs/scalar/src/filter.c:3860]: 这里没有处理完，SET FLAG之后怎么处理？

File: source/libs/scalar/src/filter.c [hunk 2]
```diff
@@ -4531,6 +4552,29 @@ static int32_t fltSclGetTimeStampDatum(SFltSclPoint *point, SFltSclDatum *d) {
   return TSDB_CODE_SUCCESS;
 }
 
+typedef struct SRewriteGetInOperContext {
+  bool hasInOper;
+} SRewriteGetInOperContext;
+
+static EDealRes rewriteInOperForTimerange(SNode **ppNode, void *pContext) {
+  SRewriteGetInOperContext *pCxt = pContext;
+  if (nodeType(*ppNode) == QUERY_NODE_OPERATOR && ((SOperatorNode *)(*ppNode))->opType == OP_TYPE_BIT_OR) {
```
Comment (dapan1121) [inline source/libs/scalar/src/filter.c]: 跟位运算有什么关系？测试没测到这里吧

File: source/libs/scalar/src/filter.c [hunk 3]
```diff
@@ -4561,8 +4605,11 @@ int32_t filterGetTimeRange(SNode *pNode, STimeWindow *win, bool *isStrict) {
         FLT_ERR_JRET(fltSclGetTimeStampDatum(endPt, &end));
         win->skey = start.i;
         win->ekey = end.i;
-        if(optNode->opType == OP_TYPE_IN) *isStrict = false;
-        else *isStrict = true;
+        if (optNode->opType == OP_TYPE_IN || hasAndTypeInOperator(info->sclCtx.node)) {
```
Comment (dapan1121) [inline source/libs/scalar/src/filter.c]: 这个逻辑不严谨

Comment (facetosea) [reply source/libs/scalar/src/filter.c]: 已修改

File: source/libs/scalar/src/filter.c (hunk 2 above)
Comment (facetosea) [reply source/libs/scalar/src/filter.c]: 修改逻辑，不在这里处理

File: source/libs/scalar/src/filter.c (hunk 1 above)
Comment (facetosea) [reply source/libs/scalar/src/filter.c:3860]: done

**Review by dapan1121 — APPROVED** [review]

**Review by guanshengliang — APPROVED** [review]

**Pull request merged**